src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package com.dev.cinema.security;
+
+import com.dev.cinema.exceptions.AuthenticationException;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.UserService;
+import com.dev.cinema.util.HashUtil;
+import java.util.Optional;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDB = userService.findByEmail(email);
+        if (userFromDB.isEmpty() || !isValid(password, userFromDB.get())) {
+            throw new AuthenticationException("Incorrect login or password");
+        }
+        return userFromDB.get();
+    }
+
+    @Override
+    public User register(String email, String password) {
+        User user = new User();
+        user.setEmail(email);
+        user.setSalt(HashUtil.getSalt());
```

Comment:
let's hash user's password in UserService

Reply:
done